src/services/codefixes/codeFixProvider.ts

```diff
@@ -0,0 +1,48 @@
+/* @internal */
+namespace ts {
+    export interface CodeFix {
+        errorCodes: number[];
+        getCodeActions(context: CodeFixContext): CodeAction[] | undefined;
+    }
+
+    export interface CodeFixContext {
+        errorCode: number;
+        sourceFile: SourceFile;
+        span: TextSpan;
+        program: Program;
+        newLineCharacter: string;
+    }
+
+    export namespace codefix {
+        const codeFixes = createMap<CodeFix[]>();
+
+        export function registerCodeFix(action: CodeFix) {
+            forEach(action.errorCodes, error => {
+                let fixes = codeFixes[error];
+                if (!fixes) {
+                    fixes = [];
+                    codeFixes[error] = fixes;
+                }
+                fixes.push(action);
+            });
+        }
+
+        export function getSupportedErrorCodes() {

```

Comment:
Yes, while we currently only return numbers for errorcodes, in the future e.g. a linter extension could prepend the errorcodes they return with a string. That way they can be differentiated and handled by separate 'fixers', even when they have the same error number.